src/alchemlyb/tests/test_units.py

```diff
@@ -0,0 +1,184 @@
+import matplotlib
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import pytest
+
+from alchemtest.gmx import load_benzene
+from alchemlyb.parsing.gmx import extract_u_nk, extract_dHdl
+from alchemlyb.preprocessing import (slicing, statistical_inefficiency,
+                                     equilibrium_detection)
+from alchemlyb.postprocessors.units import to_kcalmol, to_kT, to_kJmol
+from alchemlyb.estimators import MBAR, TI, BAR
+from alchemlyb.visualisation import plot_dF_state, plot_ti_dhdl
+
+
+def test_extract_u_nk():
+    '''Test if extract_u_nk assign the attr correctly'''
+    dataset = load_benzene()
+    u_nk = extract_u_nk(dataset['data']['Coulomb'][0], 310)
+    assert u_nk.attrs['temperature'] == 310
+    assert u_nk.attrs['energy_unit'] == 'kT'
+
+def test_extract_dHdl():
+    '''Test if extract_u_nk assign the attr correctly'''
+    dataset = load_benzene()
+    dhdl = extract_dHdl(dataset['data']['Coulomb'][0], 310)
+    assert dhdl.attrs['temperature'] == 310
+    assert dhdl.attrs['energy_unit'] == 'kT'
+
+class Test_preprocessing():
```

Comment:
add as `TestUnits` to the `test_preprocessors.py` so that in the future it is easier to add tests when anything changes in preprocessing.